app/models/Tokens.py

```diff
@@ -0,0 +1,191 @@
+        redis_db.setex(self.value, data["exp"], self.attributes.token_type)
+        self.value = None
+
+
+    async def extract_payload(self):
+        """
+        This method extracts the payload from the current token.
+        """
+
+        if self.value is None:
+            return None
+
+        # We check if the token has been revoked
+        if await self.is_revoked():
+            raise HTTPException(status_code=401, detail=CommonErrorMessages.TOKEN_REVOKED)
+
+        # We try to extract the payload from the token
+        try:
+            # We use a different key whether it is a Refresh or an Auth token.
+            # Both are supplied in the ENV variables
+            data = jwt.decode(jwt=self.value, key=str(self.attributes.secret), algorithms=[self.attributes.algorithm])
+
+        except jwt.exceptions.ExpiredSignatureError:
+            # This exception is raised if the date of the token has expired
+            raise HTTPException(status_code=401, detail=CommonErrorMessages.TOKEN_EXPIRED)
+
+        except jwt.exceptions.InvalidTokenError:
+            # This exception is raised if the token cannot be decoded.
+            raise HTTPException(status_code=401, detail=CommonErrorMessages.TOKEN_INVALID)
+
+        # We return the payload
+        return data
+
+
+    async def is_revoked(self, redis_db: redis.Redis = Redis.get_redis()):
+        """
+        This method checks if the token has been blacklisted inside Redis DB.
+        """
+        # We check if the token is revoked (in the redis DB)
+        # If the result is None, the token was not revoked
+        return not (self.value is None or redis_db.get(self.value) is None)
+
+
+class TokenPair:
+    access_token:  Token
+    refresh_token: Token
+
+    def __init__(self, access_token: str | None = None, refresh_token: str | None = None):
+        self.access_token  = Token(AvailableTokenAttributes.AUTH_TOKEN.value, access_token)
+        self.refresh_token = Token(AvailableTokenAttributes.REFRESH_TOKEN.value, refresh_token)
+
+
+    def get_tokens_in_response(self):
+        """
+        This method provides the object that needs to be returned when we want to send a new token pair.
+        """
+        return {
+            "access_token":  self.access_token.value,
+            "refresh_token": self.refresh_token.value,
+            "token_type":    "bearer"
+        }
+
+
+    def generate_tokens(self, user_id: int) -> None:
+        """
+        This method generates a pair of tokens.
+        It stores them inside the current object instance.
+        """
+        self.access_token.generate(user_id)
+        self.refresh_token.generate(user_id)
+
+
+    def revoke_tokens(self) -> None:
+        """
+        This method revokes the access token and refresh token inside the current object.
+        """
+        self.access_token.revoke()
+        self.refresh_token.revoke()
+
+
+    async def refresh_tokens(self) -> None:
```

Comment:
should be typed with Awaitable[<ReturnType>] or Coroutine[<YieldType>, <SendType>, <ReturnType>]